core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -673,7 +675,8 @@ class Partition(val topicPartition: TopicPartition,
 
       updateAssignmentAndIsr(
         assignment = partitionState.replicas.asScala.iterator.map(_.toInt).toSeq,
-        isr = Set.empty[Int],
+        followers = Seq.empty,
+        isr = Set.empty,
```

Comment:
Perhaps a separate issue, but any reason not to update ISR?

Reply:
That's a good question. Whenever the leader updates the ISR, ISR changes would not be propagated to the followers when the ZK controller is used because the leader epoch does not change. I believe that this is why we chose not to update it here because it would not reflect the real ISR. I suppose that we could revise this with KRaft because followers get all the state updates.

Reply:
Yeah, I think we should be propagating changes in the zk path as well, but that's a separate concern. I agree to leave it unchanged here in any case.